Source/WTF/wtf/text/WYHasher.h

```diff
@@ -0,0 +1,264 @@
+                    | (static_cast<uint64_t>(p[2]) << 32)
+                    | (static_cast<uint64_t>(p[3]) << 48);
+            }
+            uint64_t v;
+            memcpy(&v, p, 4);
+            v = (v | (v << 16)) & 0x0000ffff0000ffffULL;
+            return (v | (v << 8)) & 0x00ff00ff00ff00ffULL;
+        }
+    };
+
+    template<typename T, typename Reader>
+    static constexpr unsigned hash(const T* p, const unsigned characterCount)
+    {
+        auto wyr2 = Reader::wyr2;
+        auto wyr4 = Reader::wyr4;
+        auto wyr8 = Reader::wyr8;
+
+        const unsigned byteCount = characterCount << 1;
+        uint64_t seed = wymix(secret[0], secret[1]);
+        uint64_t a = 0;
+        uint64_t b = 0;
+
+        if (LIKELY(byteCount <= 16)) {
+            if (LIKELY(byteCount >= 4)) {
+                const unsigned offset = (((byteCount >> 3) << 2)) >> 1;
+                a = (wyr4(p) << 32) | wyr4(p + offset);
+                p += characterCount - 2;
+                b = (wyr4(p) << 32) | wyr4(p - offset);
+            } else if (LIKELY(byteCount > 0)) {
+                a = wyr2(p);
+                b = 0;
+            } else {
+                a = 0;
+                b = 0;
+            }
+        } else {
+            size_t i = byteCount;
+            if ((i > 48)) {
+                uint64_t see1 = seed;
+                uint64_t see2 = seed;
+                do {
+                    seed = wymix(wyr8(p) ^ secret[1], wyr8(p + 4) ^ seed);
+                    see1 = wymix(wyr8(p + 8) ^ secret[2], wyr8(p + 12) ^ see1);
+                    see2 = wymix(wyr8(p + 16) ^ secret[3], wyr8(p + 20) ^ see2);
+                    p += 24;
+                    i -= 48;
+                } while (LIKELY(i > 48));
+                seed ^= see1 ^ see2;
+            }
+            while (LIKELY(i > 16)) {
+                seed = wymix(wyr8(p) ^ secret[1], wyr8(p + 4) ^ seed);
+                i -= 16;
+                p += 8;
+            }
+            const unsigned offset = i >> 1;
+            a = wyr8(p + offset - 8);
+            b = wyr8(p + offset - 4);
+        }
+        a ^= secret[1];
+        b ^= seed;
+
+        wymum(&a, &b);
+        return wymix(a ^ secret[0] ^ byteCount, b ^ secret[1]);
+    }
+
+    template<typename T, typename Converter>
+    static unsigned computeHashImpl(const T* characters, unsigned characterCount)
+    {
+        ASSERT(sizeof(decltype(Converter::convert(std::declval<T>()))) == 2);
+        Vector<uint16_t> buffer;
+        for (size_t i = 0; i < characterCount; ++i)
+            buffer.append(Converter::convert(*characters++));
+        return hash<uint16_t, Reader16Bit<uint16_t>>(buffer.data(), characterCount);
+    }
+
+    template<typename T>
+    static constexpr unsigned computeHashImpl(const T* characters, unsigned characterCount)
+    {
+        if constexpr (sizeof(T) == 2)
+            return hash<T, Reader16Bit<T>>(characters, characterCount);
```

Comment:
Let's use
```
if constexpr (sizeof(T) == 2)
    return hash<T, Reader16Bit<T>>(characters, characterCount);
else
    return hash<T, Reader8Bit<T>>(characters, characterCount);
```
otherwise, else side does not become `constexpr`.